site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -277,6 +277,34 @@ public boolean filterUserSites(List<SiteDTO> sitesReturnObject,
 
     }
 
+    @Override
+    public List<SiteNameDTO> findAssignedSites() {
```

Comment:
check if there are roles assigned to the user - if not throw error
check if there are sites - if not throw error
then convert site objects into siteName objects containing siteName and siteId.